### PR TITLE
*: Fix bug for select @@character_set_database

### PR DESCRIPTION
**Pull request by shenli**
1. Add charset info when create schema
2. Set charset variable when run use statement
Fix: https://github.com/pingcap/tidb/issues/706

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -240,11 +240,21 @@ func (d *ddl) CreateSchema(ctx context.Context, schema model.CIStr) (err error)
...
+		collation = charsetInfo.Col
+	} else {
+		charsetName, collation = getDefaultCharsetAndCollate()
+	}
 
 	job := &model.Job{
 		SchemaID: schemaID,
 		Type:     model.ActionCreateSchema,
-		Args:     []interface{}{schema},
+		Args:     []interface{}{schema, charsetName, collation},
```
Comment (ngaut) [inline ddl/ddl.go]: It's hard to maintain information like that, please use structure.

File: stmt/stmts/use_test.go [hunk 2]
```diff
@@ -44,3 +44,39 @@ func (s *testStmtSuite) TestUse(c *C) {
...
+	testSQL = `create database if not exists cd_test_latin1 CHARACTER SET latin1 COLLATE latin1_swedish_ci;`
+	mustExec(c, s.testDB, testSQL)
+
+	testSQL = `use cd_test_utf8;`
+	mustExec(c, s.testDB, testSQL)
+
+	tx := mustBegin(c, s.testDB)
+	rows, err := tx.Query(`select @@character_set_database;`)
+	c.Assert(err, IsNil)
+	matchRows(c, rows, [][]interface{}{{"utf8"}})
```
Comment (siddontang) [inline stmt/stmts/use_test.go]: matchRows called rows.Close inner, no need to call below again.

**Comment by shenli** [comment]
PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
But u have to merge master to pass CI to suit for the go-hbase package.

**Comment by shenli** [comment]
PTAL

File: model/model.go [hunk 3]
```diff
@@ -142,7 +142,7 @@ type DBInfo struct {
 	Name    CIStr        `json:"db_name"` // DB name.
 	Charset string       `json:"charset"`
 	Collate string       `json:"collate"`
-	Tables  []*TableInfo `json:"-"` // Tables in the DB.
+	Tables  []*TableInfo `json:"tables"` // Tables in the DB.
```
Comment (siddontang) [inline model/model.go]: we don't need to marshal tables in database meta.

**Comment by shenli** [comment]
@siddontang PTAL

**Comment by shenli** [comment]
➜  tidb git:(shenli/issue-706) ✗ make ddl_test
$(which godep) go test ./ddl/... -skip_ddl=false
ok  	github.com/pingcap/tidb/ddl	70.280s

**Comment by siddontang** [comment]
LGTM

**Pull request merged**